File: ci/templates/go-feature/internal/services/feature_service.go
package services

import (
	"context"
	"example.com/feature/internal/models"
	"example.com/feature/internal/store"
)

type FeatureService struct {
	store store.FeatureStore
}

func NewFeatureService(s store.FeatureStore) *FeatureService {
	return &FeatureService{store: s}
}

func (s *FeatureService) GetOrCreate(ctx context.Context, id, name string) (*models.Feature, error) {
	existing, err := s.store.Get(ctx, id)
	if err != nil {
		return nil, err
	}
	if existing != nil {
		return existing, nil
	}
	f := &models.Feature{ID: id, Name: name}
	if err := s.store.Save(ctx, f); err != nil {
		return nil, err
	}
	return f, nil
}

File: ci/templates/js-feature/src/modules/feature/featureService.ts
export interface Feature {
  id: string;
  name: string;
}

const store = new Map<string, Feature>();

export function getOrCreate(id: string, name: string): Feature {
  const existing = store.get(id);
  if (existing) return existing;
  const f: Feature = { id, name };
  store.set(id, f);
  return f;
}

export function clearStore() {
  store.clear();
}
